File: src-tauri/src/commands/plugins.rs
use crate::scanners::plugins::{scan_all_plugins, PluginScanResult};

#[tauri::command]
pub fn scan_plugins(workspace_path: Option<String>) -> Result<PluginScanResult, String> {
    scan_all_plugins(workspace_path.as_deref())
}

File: src/lib/api/plugins.ts
import { invoke } from "@tauri-apps/api/core";
import type { PluginScanResult } from "@/types";

/**
 * Scan installed plugins and marketplace catalogs across
 * Claude Code CLI, Claude Desktop Cowork, and Gemini CLI.
 */
export async function scanPlugins(
  workspacePath?: string
): Promise<PluginScanResult> {
  return invoke<PluginScanResult>("scan_plugins", { workspacePath });
}
